src/types/pokemonType.ts

```diff
@@ -0,0 +1,48 @@
+export interface IPokemonData {
+  id: number;
+  name: string;
+  isOwned: boolean;
+  description: string;
+  image: string;
+  profile: Profile | null;
+  baseStats: BaseStats | null;
```

Comment:
profile?: Profile;
  baseStats?: BaseStats;

Reply:
Added the "?" but cant remove the " | null".
Yuval said to leave it as it is.

If i remove i get this error:
Type '({ profile: { pokemon_id: number; height: string; weight: string; ability: string[]; types: string[]; } | null; baseStats: { pokemon_id: number; hp: number; attack: number; defense: number; sp_attack: number; sp_defense: number; speed: number; power_level: number; } | null; } & { ...; })[]' is not assignable to type 'IPokemonData[]'.
  Type '{ profile: { pokemon_id: number; height: string; weight: string; ability: string[]; types: string[]; } | null; baseStats: { pokemon_id: number; hp: number; attack: number; defense: number; sp_attack: number; sp_defense: number; speed: number; power_level: number; } | null; } & { ...; }' is not assignable to type 'IPokemonData'.
    Types of property 'profile' are incompatible.
      Type '{ pokemon_id: number; height: string; weight: string; ability: string[]; types: string[]; } | null' is not assignable to type 'Profile | undefined'.
        Type 'null' is not assignable to type 'Profile | undefined'.